src/System.ServiceModel.Federation/src/System/ServiceModel/Federation/WSTrustChannelFactory.cs

```diff
@@ -0,0 +1,90 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.InteropServices;
+using System.ServiceModel.Channels;
+using System.ServiceModel.Description;
+using Microsoft.IdentityModel.Protocols.WsTrust;
+
+namespace System.ServiceModel.Federation
+{
+    /// <summary>
+    /// A <see cref="WSTrustChannelFactory" /> that produces <see cref="WSTrustChannel" /> objects used to communicate with a WS-Trust endpoint.
+    /// </summary>
+    [ComVisible(false)]
+    public class WSTrustChannelFactory : ChannelFactory<IWSTrustChannelContract>
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="WSTrustChannelFactory" /> class with a specified endpoint.
+        /// configuration name.
+        /// </summary>
+        /// <param name="endpointConfigurationName">The configuration name used for the endpoint.</param>
+        public WSTrustChannelFactory(string endpointConfigurationName)
+            : base(endpointConfigurationName)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="WSTrustChannelFactory" /> class with a specified endpoint.
+        /// </summary>
+        /// <param name="endpoint">The <see cref="ServiceEndpoint" />for the channels produced by the factory.</param>
+        public WSTrustChannelFactory(ServiceEndpoint endpoint)
+            : base(endpoint)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="WSTrustChannelFactory" /> class associated with a specified
+        /// name for the endpoint configuration and remote address.
+        /// </summary>
+        /// <param name="endpointConfigurationName">The configuration name used for the endpoint.</param>
+        /// <param name="remoteAddress">The <see cref="EndpointAddress" /> that provides the location of the service.</param>
+        public WSTrustChannelFactory(string endpointConfigurationName, EndpointAddress remoteAddress)
+            : base(endpointConfigurationName, remoteAddress)
```

Comment:
Same for this one, it will just throw so remove.

Reply:
addressed in: ab9b3c1cb00f558dcd6ac6f3b02663ace5e25750